src/utils/inverse_matrix.cpp

```diff
@@ -0,0 +1,45 @@
+//========================================================================================
+// Athena++ astrophysical MHD code
+// Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+
+// Athena headers
+#include "../athena.hpp"
+#include "./utils.hpp"
+
+
+//======================================================================================
+//! \file inverse_matrix.cpp
+//======================================================================================
+
+
+//--------------------------------------------------------------------------------------
+//! \fn  void  InverseMatrix(int n, AthenaArray<Real> &a, AthenaArray<Real> &b)
+// \brief Inverse matrix solver
+// a: input matrix; n: matrix size, b: return matrix
+// Note: the input matrix will be DESTROYED
+
+void InverseMatrix(int n, AthenaArray<Real> &a, AthenaArray<Real> &b) {
+  AthenaArray<int> indx;
+  AthenaArray<Real> col;
+  Real d;
+
+  indx.NewAthenaArray(n);
+  col.NewAthenaArray(n);
```

Comment:
If this routine is frequently used, it may be better to pre-allocate these arrays as array allocation is generally expensive. You can declare these arrays within this fine using anonymous namespace, and check if they are allocated using AthenaArray::IsAllocated (or GetDim1).

Reply:
All these functions related to matrix are used only in one place. In most applications, they may not be used at all.